lib/api/controllers/baseController.js

```diff
@@ -92,6 +94,229 @@ class BaseController {
+  }
+
+  /**
+   * Extracts an object parameter from the request input
+   *
+   * @param {Request} request - Request object
+   * @param {String} paramPath - Path of the parameter to extract (eg: 'body.foo')
+   * @param {?String[]} defaultValue
+   *
+   * @throws
+   * @returns {String[]}
+   */
+  getObjectParam(request, paramPath, defaultValue = null) {
+    const objectParam = _.get(request.input, paramPath, defaultValue);
+
+    if (objectParam === null) {
+      errorsManager.throw('missing_param', paramPath);
+    }
+
+    if (! _.isPlainObject(objectParam)) {
+      errorsManager.throw('invalid_param_type', paramPath, objectParam, 'Object');
+    }
+
+    return objectParam;
+  }
+
+  /**
+   * Returns the index specified in the request
+   *
+   * @param {Request} request
+   *
+   * @throws
+   * @returns {String}
+   */
+  getIndex (request) {
+    const index = request.input.resource.index;
+
+    if (! index) {
+      errorsManager.throw('missing_index');
+    }
+
+    return index;
+  }
+
+  /**
+   * Returns the index and collection specified in the request
+   *
+   * @param {Request} request
+   *
+   * @throws
+   * @returns {Object} { index, collection }
+   */
+  getIndexAndCollection (request) {
+    const
+      index = this.getIndex(request),
+      collection = request.input.resource.collection;
+
+    if (! collection) {
+      errorsManager.throw('missing_collection');
+    }
+
+    return { index, collection };
+  }
+
+  /**
+   * Returns the id specified in the request
+   *
+   * @param {Request} request
+   *
+   * @throws
+   * @returns {String}
+   */
+  getId (request) {
+    const id = request.input.resource._id;
+
+    if (! id) {
+      errorsManager.throw('missing_id');
+    }
+
+    if (typeof id !== 'string') {
+      errorsManager.throw('wrong_id_type');
+    }
+
+    if (id.charAt(0) === '_') {
+      errorsManager.throw('wrong_id_format');
+    }
+
+    return id;
+  }
+
+  /**
+   * Returns the current user id
+   *
+   * @param {Request} request
+   *
+   * @returns {String}
+   */
+  getUserId (request) {
+    return _.get(request, 'context.user._id', null);
+  }
+
+  getSearchParams (request) {
+    const
+      from = this.getNumberParam(request, 'args.from', 0),
+      size = this.getNumberParam(request, 'args.size', 10),
+      scroll = this.getScrollParam(request, 'args.scroll', undefined),
+      query = this.getObjectParam(request, 'body.query', {});
+
+    return { from, size, scroll, query };
+  }
+
+  /**
+   * Throws if the body contain one of the specified attribute
+   *
+   * @param {Request} request
+   * @param  {...any} paths
+   */
+  assertBodyHasNotAttributes (request, ...paths) {
+    for (const path of paths) {
+      if (_.get(request.input.body, path)) {
+        errorsManager.throw('must_not_specify_body_attribute', path);
+      }
+    }
+  }
+
+  /**
+   * Throws if the strategy does not exists
+   *
+   * @todo move this method in some kind of "Security" class
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)